### ISSUE TITLE
Make timing on `package-refresh-contents` configurable.

### ISSUE THREAD

**Issue by jeffbowman** [SystemCrafters/crafted-emacs#259]
From this conversation on Discord:

njheyu — 12/09/2022 7:41 AM
Does crafted emacs enable automatic package update(package refresh) by default? Where can I disable this behaviour?

WolfJB — 12/09/2022 9:07 AM
Sort of. Crafted Emacs will check to see if the archives are stale on startup. If they are out of date by 24 hours, they will be synchronized asynchronously.  However, that only occurs on startup.  If you leave Emacs running for long periods (several days), the package archives will become stale if you don't manually run (package-refresh-contents)
You can take a look at the implementation in bootstrap/crafted-package-bootstrap.el file where you clone Crafted Emacs

njheyu — 12/09/2022 10:50 AM
Thank you! Is there a way to change that 24 hours to one month? I am asking because I use it for my work, but our company has a firewall by default so it complains if I don’t run it with proxy. I want to only run with proxy once in a month, but my lisp is not good enough to make such a modification to the code. Appreciate your help! 

WolfJB — 12/09/2022 11:13 AM
hmm... not currently an easy way, I'll have to work on it a bit.  To get you a solution, in that file, replace the body of crafted-package-archive-stale-p with just the word nil  ala:

(defun crafted-package-archive-stale-p (archive) nil) 

which basically says to always return "false" when the check occurs to see if any of the archives are stale.  You'll then have to manually update the archives any time you want to install packages.  In the meanwhile, I'll work on a solution to make the duration more configurable.

njheyu — 12/09/2022 12:55 PM
Sounds great! Thank you so much!

**Comment by jeffbowman** [comment]
Potential resolution provided, waiting for testing - as reported on Discord

WolfJB — Today at 1:51 PM
@njheyu  I have made a few updates to Crafted Emacs, they are on the branch package-contents-timing, if you could pull from that branch and test to make sure things are working for you, that would be great.  Please review the docstrings for the crafted-bootstrap-package-perform-stale-archive-check (boolean, t is the default, you may wish to switch that to nil). and also crafted-bootstrap-package-update-days (number, defaults to 1, if you leave the prior variable set to t, you  may wish to set this value to a higher number).  Assuming this works for you, I'll merge it to master.

**Comment by jeffbowman** [comment]
Response from Discord:

>njheyu — Today at 1:59 PM
I tested today and it works great! If I don’t set the variable I get an error due to the firewall, and if I set the variable to 30 in early-config.el then it no longer gives me the error. Thanks a lot for the fast turnaround!
WolfJB — Today at 8:47 PM
Thanks for testing, I'll merge that to master in the next day or so, I need to update the docs to cover that change.
